var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -62,50 +63,73 @@ class Mfem(Package):
+    variant('suite-sparse', default=False,
+        description='Enable serial, sparse direct solvers')
+    variant('petsc', default=False,
+        description='Enable PETSc solvers, preconditioners, etc..')
+    variant('sundials', default=False,
+        description='Enable Sundials time integrators')
+    variant('mpfr', default=False,
+        description='Enable precise, 1D quadrature rules')
+    variant('lapack', default=False,
+        description='Use external blas/lapack routines')
+    variant('debug', default=False,
+        description='Build debug instead of optimized version')
+    variant('netcdf', default=False,
+        description='Enable Cubit/Genesis reader')
+    variant('gzstream', default=True,
+        description='Support zip\'d streams for I/O')
+    variant('examples', default=False,
+        description='Build and install examples')
+    variant('minapps', default=False,
+        description='Build and install minapps')
 
     depends_on('blas', when='+lapack')
+    depends_on('blas', when='+suite-sparse')
     depends_on('lapack', when='+lapack')
+    depends_on('lapack', when='+suite-sparse')
 
     depends_on('mpi', when='+mpi')
-    depends_on('metis', when='+mpi')
-    depends_on('hypre', when='+mpi')
-
-    depends_on('hypre', when='+hypre')
-
-    depends_on('metis@4:', when='+metis')
-
-    depends_on('suite-sparse', when='+suite-sparse')
-    depends_on('blas', when='+suite-sparse')
-    depends_on('lapack', when='+suite-sparse')
+    depends_on('metis')
+    depends_on('parmetis', when='+superlu-dist')
+    depends_on('metis@5:', when='+superlu-dist')
     depends_on('metis@5:', when='+suite-sparse ^suite-sparse@4.5:')
-    depends_on('cmake', when='^metis@5:', type='build')
+    depends_on('hypre~internal-superlu', when='+mpi')
+    depends_on('hypre@develop~internal-superlu', when='+petsc +hypre')
 
+    depends_on('sundials@2.7:+hypre', when='+sundials')
+    depends_on('suite-sparse', when='+suite-sparse')
     depends_on('superlu-dist', when='@3.2: +superlu-dist')
+    depends_on('petsc@develop', when='+petsc')
 
+    depends_on('mpfr', when='+mpfr')
+    depends_on('cmake', when='^metis@5:', type='build')
     depends_on('netcdf', when='@3.2: +netcdf')
     depends_on('zlib', when='@3.2: +netcdf')
     depends_on('hdf5', when='@3.2: +netcdf')
+    depends_on('libunwind', when='+debug')
+    depends_on('zlib', when='+gzstream')
 
     patch('mfem_ppc_build.patch', when='@3.2:3.3 arch=ppc64le')
 
     def check_variants(self, spec):
-        if '+mpi' in spec and ('+hypre' not in spec or '+metis' not in spec):
-            raise InstallError('mfem+mpi must be built with +hypre ' +
-                               'and +metis!')
-        if '+suite-sparse' in spec and ('+metis' not in spec or
-                                        '+lapack' not in spec):
+        if '+mpi' in spec and '+hypre' not in spec:
+            raise InstallError('mfem+mpi must be built with +hypre')
```

Comment:
You should replace all of the logic in this `check_variants` function with `conflicts()` directives. The former will crash at build time whereas the latter will crash at concretization. The syntax looks like:
```python
conflicts('+mpi', when='~hypre')
```